db/blip_handler_collections.go

```diff
@@ -70,11 +70,15 @@ func (bh *blipHandler) handleGetCollections(rq *blip.Message) error {
 		}
 		db = &Database{DatabaseContext: collection.CollectionCtx}
 
-		value, err := db.GetSpecial(DocTypeLocal, CheckpointDocIDPrefix+requestBody.CheckpointIDs[i])
+		key := CheckpointDocIDPrefix + requestBody.CheckpointIDs[i]
+		value, err := db.GetSpecial(DocTypeLocal, key)
 		if err != nil {
-			// handle key not found here specially
-			// this will return key "_sync:local:checkpoint/id" missing if collection exists but not present
-			checkpoints[i] = Body{}
+			status, _ := base.ErrorAsHTTPStatus(err)
+			if status == http.StatusNotFound {
+				checkpoints[i] = Body{}
+			} else {
+				checkpoints[i] = nil
```

Comment:
As discussed, let's keep this PR moving but revisit error handling here in CBG-2203 

Could add the TODO/Warning log until then.

```suggestion
				// TODO: CBG-2203 - should we return an error such that the client disconnects here?
				base.WarnfCtx(bh.loggingCtx, "Unable to fetch client checkpoint %q for collection %v", key, collection)
				checkpoints[i] = nil
```